File: path/to/LocaleProvider.js
import { getSavedLocale } from './settings'; // 假设有一个函数可以获取保存的语言设置

const defaultLocale = getSavedLocale() || 'en'; // 使用保存的语言设置，默认值为 'en'

export const LocaleProvider = ({ children }) => {
    const [locale, setLocale] = useState(defaultLocale);
    // ...
} 

File: lib/providers/locale_provider.dart
import 'package:flutter/material.dart';
import '../services/settings_service.dart';
import '../services/service_locator.dart';

class LocaleProvider extends ChangeNotifier {
  Locale _locale = const Locale('zh'); // 设置默认值
  final ISettingsService _settingsService = getIt<ISettingsService>();

  LocaleProvider() {
    _initLocale();
  }

  Future<void> _initLocale() async {
    try {
      final savedLanguage = await _settingsService.getLanguage();
      _locale = Locale(savedLanguage);
      notifyListeners();
    } catch (e) {
      debugPrint('Error loading language settings: $e');
      // 保持默认值 'zh'
    }
  }

  Locale get locale => _locale;

  void setLocale(Locale locale) {
    _locale = locale;
    notifyListeners();
  }
}
